src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs

```diff
@@ -91,7 +91,7 @@ public static byte[] sqlite3_column_blob(Sqlite3StmtHandle pStmt, int iCol)
             var ptr = Sqlite3.column_blob(pStmt, iCol);
             if (ptr == IntPtr.Zero)
             {
-                return null;
+                return new byte[0];
```

Comment:
Boo. I hate #if-in just to use Array.Empty. If caching is really an issue, cache it statically on this type.